File: tests/fixtures/realistic/go_multihop_sanitized/service.go
// Middle helper (fileB) for the go_multihop_sanitized fixture.
//
// Unlike the positive fixture, loadFile() runs the tainted value through
// filepath.Clean() — a configured sanitizer for go/taint-path-traversal —
// before forwarding it to readData(). The sanitizer collapses the value to
// "clean", so the composed summary must NOT record a params_to_sink flow, and
// the chain breaks: no taint finding on a directory scan.

package go_multihop

import "path/filepath"

func loadFile(name string) []byte {
	safe := filepath.Clean(name)
	return readData(safe)
}

File: tests/fixtures/realistic/java_multihop_broken/Service.java
// Middle helper (fileB) for the java_multihop_broken fixture.
//
// Unlike the positive fixture, process() does NOT forward its tainted
// parameter. It replaces the value with a constant before calling
// QueryHelper.runQuery(), so the composed summary must NOT record a
// params_to_sink flow and the chain breaks: no taint finding on a directory
// scan. (Java ships no configured sanitizer rules, so a clean reassignment
// stands in for the sanitizer-call break used by the other languages.)

class Service {
    static void process(String term) throws Exception {
        String safe = "constant";
        QueryHelper.runQuery(safe);
    }
}
